include/revng/TupleTree/TupleTree.h

```diff
@@ -1129,6 +1130,22 @@ void serialize(S &Stream, const T &Element) {
   serialize(Stream, const_cast<T &>(Element));
 }
 
+template<Yamlizable TupleLike>
+llvm::Error
+serializeToFile(const TupleLike &Tuple, const llvm::StringRef &Path) {
```

Comment:
Why `loadFromFile` is a method and this is not? What are these going to be used for?

Reply:
i saw that the serialize routine was not already here and i put serializedToFile here and i saw the fromString method in the other place and i added fromFile there, i can move them around tho.

they will be used to load and save the model on disk

Reply:
What I meant is: if you don't see a use case for `serializeToFile` `TupleLike`s in general, but you'll be using it on `TupleTree<T>` only, make this function a method of `TupleTree<T>`, just as `fromFile`.

Reply:
a issue with this is that if someone holds a reference to a binary instead of a reference to a tuple tree binary it will not be able to serialize it, while with the current version it will be able to

Reply:
and due to compatibility with load model pass we were returning a const binary& to people that were requiring a readonly model